pkg/R/sparkRBackend.R

```diff
@@ -72,18 +72,21 @@ invokeJava <- function(isStatic, objId, methodName, ...) {
   writeInt(rc, length(args))
   writeArgs(rc, args)
 
+  # Construct the whole request message to send it once,
+  # avoiding write-write-read pattern in case of Nagle's algorithm
   bytesToSend <- rawConnectionValue(rc)
+  close(rc)
+  rc <- rawConnection(raw(0), "r+")
+  writeInt(rc, length(bytesToSend))
```

Comment:
I found reusing the variables a bit contrived. It's okay to reuse `rc`, but let's name the two payloads (the original `bytesToSend` and the final message `bytesToSend`) to different things?